### PR TITLE
Refactor remove actionitem

### PR DESCRIPTION
**Pull request by QizhengMo**

### CODE REVIEW

File: arex-schedule-web-api/src/test/java/com/arextest/schedule/mockImpl/MockMultiVersionExecutionImpl.java
Comment (QizhengMo) [inline arex-schedule-web-api/src/test/java/com/arextest/schedule/mockImpl/MockMultiVersionExecutionImpl.java:1]: Test only

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/service/ReplayCaseTransmitService.java [hunk 1]
```diff
@@ -55,61 +58,73 @@ public class ReplayCaseTransmitService {
...
-                progressTracer.finishCaseByPlan(replayPlan);
-            }
-        } else {
-            for (int i = 0; i < contextCasesCount; i++) {
-                progressTracer.finishCaseByAction(replayActionItem);
+            // if we skip the rest of cases remaining in the action item, set its status
+            if (replayActionItem.getReplayCaseCount() == replayActionItem.getCaseProcessCount().intValue()) {
+                progressEvent.onActionInterrupted(replayActionItem);
+                for (int i = 0; i < contextCasesCount; i++) {
+                    progressTracer.finishCaseByPlan(replayPlan);
```
Comment (wildeslam) [inline arex-schedule-web-api/src/main/java/com/arextest/schedule/service/ReplayCaseTransmitService.java:113]: Why does It need to repeat the same operation?

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/service/PlanConsumeService.java [hunk 2]
```diff
@@ -151,111 +147,45 @@ private void consumePlan(ReplayPlan replayPlan) {
...
-                }
-                BizLogger.recordContextProcessedNormal(executionContext, replayActionItem, contextCount);
+        int contextCount = 0;
+        List<ReplayActionCaseItem> caseItems = Collections.emptyList();
+        while (true) {
+            // checkpoint: before sending page of cases
+            if (executionStatus.isAbnormal()) {
+                break;
+            }
+            ReplayActionCaseItem lastItem = caseItems.size() > 0 ? caseItems.get(caseItems.size() - 1) : null;
```
[the remark this answers is not in the document]
Comment (QizhengMo) [reply arex-schedule-web-api/src/main/java/com/arextest/schedule/service/PlanConsumeService.java:175]: the query result is ordered by _id, 
for example the previouse result is _id : 1,2,3,4,5
we need to query the next batch by _id > 5

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/service/ReplayCaseTransmitService.java (hunk 1 above)
Comment (QizhengMo) [reply arex-schedule-web-api/src/main/java/com/arextest/schedule/service/ReplayCaseTransmitService.java:113]: each call to the progressTracer.finishCaseByPlan() method will reduce the redis plan case counter by 1,
if we skip 100 case, we will need to call it 100 times,
Batch redis operation is not currently supported by Arex-Common package

**Review by mr3 — APPROVED** [review]

**Review by coryhh — APPROVED** [review]

**Pull request merged**